index.js

```diff
@@ -2,7 +2,7 @@ function buildThenable() {
   return {
     then: function(onFulfill, onReject) {
       try {
-        if (this.resolved) {
+        if (this.resolved && !this.rejected) {
```

Comment:
My guess would be that we call `resolves` in one `beforeEach` and then `rejects` in another. We really need to make our resolves/rejects able to be dependent on the arguments to the stub (ie. make `withArgs` work for stubbed promises). That would let us do conditional behavior based on args, which would move us to more of a stubbing pattern vs. mocking, and would probably eliminate scenarios where we do both of these.